prime-router/src/test/kotlin/azure/SendFunctionTests.kt

```diff
@@ -134,6 +175,68 @@ class SendFunctionTests {
         assertThat(nextEvent!!.retryToken).isNull()
     }
 
+    @Test
+    fun `Test with null transport`() {
+        var nextEvent: ReportEvent? = null
+        val reportList = listOf(reportFile)
+        setupLogger()
+        setupWorkflow()
+        every { workflowEngine.handleReportEvent(any(), any()) }.answers {
+            val block = secondArg() as
+                    (header: WorkflowEngine.Header, retryToken: RetryToken?, txn: Configuration?) -> ReportEvent
+            val header = makeIgnoreDotHL7NullHeader()
+            nextEvent = block(header, null, null)
+        }
+        every { nullTransport.send(any(), any(), any(), any(), any(), any(), any(), any(), any()) }.returns(null)
+        every { workflowEngine.recordAction(any()) }.returns(Unit)
+        every { workflowEngine.azureEventService.trackEvent(any()) }.returns(Unit)
+        every { workflowEngine.reportService.getRootReports(any()) } returns reportList
+        every { workflowEngine.db } returns mockk<DatabaseAccess>()
+        mockkObject(Report.Companion)
+        every { Report.formExternalFilename(any(), any(), any(), any(), any(), any(), any()) } returns ""
+
+        // Invoke
+        val event = ReportEvent(Event.EventAction.SEND, reportId, false)
+        SendFunction(workflowEngine).run(event.toQueueMessage(), context)
+
+        // Verify
+        verify { nullTransport.send(any(), any(), any(), any(), any(), any(), any(), any(), any()) }
```

Comment:
I'm still learning my junit/kotlin unit testing. I suspect this verify isn't verifying anything? If we want to verify that the send method was called on nullTransport I think it needs something more, but totally not sure.

Reply:
My understanding is that's exactly what verify is for.... I could be wrong, tho

Reply:
Yup this was noobie confusion and I didn't take the time to look it up myself because I was rushed yesterday. Our verify statements, and mocking in general, uses the `MockK` library for Kotlin. This blog is a pretty good overview of mocking using it: https://notwoods.github.io/mockk-guidebook/docs/mocking/verify/

> Using verify to verify that a function was called looks a lot like using every for stubbing. A simple example is to call a method and immediately check that it was called.